File: frontend/src/types/history.ts
export interface HistoryItem {
  examId: string;

  questionBank: string;

  mode: string;

  questionCount: number;

  correct: number;

  wrong: number;

  unanswered: number;

  percentage: number;

  completedAt: string;

  startedAt: string;

  timed: boolean;

  durationMinutes: number | null;
}

export interface HistoryListResponse {
  exams: HistoryItem[];
}

File: backend/schemas/history.py
from pydantic import BaseModel


class HistoryItem(BaseModel):
    examId: str

    questionBank: str

    mode: str

    questionCount: int

    correct: int

    wrong: int

    unanswered: int

    percentage: float

    completedAt: str

    startedAt: str

    timed: bool

    durationMinutes: int | None = None


class HistoryListResponse(BaseModel):
    exams: list[HistoryItem]


class DeleteHistoryResponse(BaseModel):
    success: bool